qemu/tests/win_virtio_driver_installer_repair.py

```diff
@@ -19,106 +17,72 @@ def run(test, params, env):
        Verify target driver.
-    8) Run virtio-win-guest-tools.exe repair test by uninstall
-       the target driver.
-    9) Run the driver function test after virtio-win-guest-tools.exe repair.
-    10) Repeat step 9 and 10 for other drivers.
+    5) Uninstall dirvers by msi uninstall.
+    6) Run virtio-win-guest-tools.exe repair test.
+    7) Run the driver function test after repair.
 
     :param test: QEMU test object
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment
     """
-    devcon_path = params["devcon_path"]
     run_install_cmd = params["run_install_cmd"]
-    installer_pkg_check_cmd = params["installer_pkg_check_cmd"]
-
-    # gagent version check test config
-    qemu_ga_pkg = params["qemu_ga_pkg"]
-    gagent_pkg_info_cmd = params["gagent_pkg_info_cmd"]
-    gagent_install_cmd = params["gagent_install_cmd"]
-    gagent_uninstall_cmd = params["gagent_uninstall_cmd"]
+    run_repair_cmd = params["run_repair_cmd"]
 
     vm = env.get_vm(params["main_vm"])
     session = vm.wait_for_login()
 
-    expected_gagent_version = win_driver_installer_test.install_gagent(
-        session, test,
-        qemu_ga_pkg,
-        gagent_install_cmd,
-        gagent_pkg_info_cmd)
-    win_driver_installer_test.uninstall_gagent(session, test,
-                                               gagent_uninstall_cmd)
     win_driver_installer_test.win_uninstall_all_drivers(session,
                                                         test, params)
     session = vm.reboot(session)
-    win_driver_installer_test.install_test_with_screen_on_desktop(
-        vm, session, test,
+
+    test.log.info("Install virtio-win driver by installer.")
+    session = win_driver_installer_test.run_installer_with_interaction(
+        vm, session, test, params,
         run_install_cmd,
-        installer_pkg_check_cmd,
         copy_files_params=params)
-    win_driver_installer_test.win_installer_test(session, test, params)
-    win_driver_installer_test.check_gagent_version(session, test,
-                                                   gagent_pkg_info_cmd,
-                                                   expected_gagent_version)
+
     win_driver_installer_test.driver_check(session, test, params)
 
     error_context.context("Run virtio-win-guest-tools.exe repair test",
                           test.log.info)
-    unrepaired_driver = []
+    test.log.info("Remove virtio-win driver by msi.")
+    session = win_driver_utils.remove_driver_by_msi(session, vm, params)
+
+    test.log.info("Repair virtio-win driver by installer.")
+    session = win_driver_installer_test.run_installer_with_interaction(
+        vm, session, test, params,
+        run_repair_cmd)
+
+    # driver check after repair
+    win_driver_installer_test.driver_check(session, test, params)
+    # start viofs test after repair
+    virtio_fs_utils.run_viofs_service(test, params, session)
```

Comment:
@xiagao Previously this only run for virtiofs driver function test now moved up so it will be run before all the test really start, then if this step failed all the other drivers' test could be skipped, is that expected and/or accepted?

Reply:
Yes, it makes sense, I'll move it into the loop and add condition for it. Thanks.

Reply:
@vivianQizhu Updated.